src/utils/__tests__/merge-nodes-with-matching-flavors.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { format } from 'prettier';
+
+import { getCodeFromAst } from '../get-code-from-ast';
+import { getImportNodes } from '../get-import-nodes';
+import { getSortedNodes } from '../get-sorted-nodes';
+
+it('should merge duplicate imports within a given chunk', () => {
+    const code = `
+    import type { A } from 'a';
+    import { Junk } from 'junk-group-1'
+    import type { B } from 'a';
+    import "./side-effects1";
+    // C, E and D will be separated from A, B because side-effects in-between
+    import type { C } from 'a';
+    import { D } from "a";
+    import type { E } from "a";
+    // prettier-ignore
+    import type { NoMerge1 } from "a";
+    // prettier-ignore
+    import { NoMerge2 } from "a";
+    import { H } from 'b';
+    import { F } from 'a';
+    // F Will be alone because prettier-ignore in-between
+
+    import { G } from 'b';
+    import * as J from 'c';
+    import { Junk2 } from 'junk-group-2'
+    import * as K from "c";
+    // * as J, * as K can't merge because both Namespaces
+    import {I} from "c"
+    import { default as Def2 } from 'd';
+    import { default as Def1 } from 'd';
+    import Foo1 from 'e';
+    import Foo2 from 'e';
```

Comment:
There's a ton going on here, it might make the test more readable if it were broken into separate test cases.

Reply:
More granular unit tests were submitted in #20